src/monster.cpp

```diff
@@ -2236,6 +2253,15 @@ void monster::drop_items_on_death()
         items = remaining;
     }
 
+    // Adds all items in the monsters inventory to be dropped.
+    // TODO: Apply random damage to items ? Modify them in some way ?
+    if( !inv.empty() )
```

Comment:
This unnecessary. Iterating over an empty container is just fine.

And you don't even need to iterate at all:
```suggestion
    items.insert(items.end(), inv.begin(), inv.end());
```